editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs

```diff
@@ -15,39 +15,35 @@ use graphene_core::renderer::format_transform_matrix;
 use glam::{DAffine2, DVec2};
 use serde::{Deserialize, Serialize};
 
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+enum TransformOperation {
+	None,
+	Pan { pre_commit_pan: DVec2 },
+	Rotate { pre_commit_tilt: f64, snap_tilt: bool, snap_tilt_released: bool },
+	Zoom { pre_commit_zoom: f64, snap_zoom_enabled: bool },
+}
+
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub struct NavigationMessageHandler {
 	pub pan: DVec2,
-	panning: bool,
-	snap_tilt: bool,
-	snap_tilt_released: bool,
-
 	pub tilt: f64,
-	tilting: bool,
-
 	pub zoom: f64,
-	zooming: bool,
-	snap_zoom: bool,
 
+	transform_operation: TransformOperation,
 	mouse_position: ViewportPosition,
+	dispatched_from_menu: bool,
```

Comment:
Thoughts on renaming this to something like `finish_operation_with_click`?

Reply:
maybe a little vague? it finished with a click no matter where it is dispatched from. `blocking_mouse_events` be more specific, it's what it's doing not why it's doing it.

Reply:
It doesn't really finish with a click if you are dragging the middle mouse button - it finishes with mouse up.

Reply:
okay I see your point I'll change it.